From f0c25cd128a01c3805207b83d422149b170e88a8 Mon Sep 17 00:00:00 2001
From: electimon <electimon@gmail.com>
Date: Tue, 1 Apr 2025 14:00:10 -0400
Subject: [PATCH 54/55] HintManagerService: Generate dummy mSupportInfo for non
 power AIDL devices

* Before it would generate dummy info when the IPower AIDL version
  was too old. Now we will generate it when IPower is just completely
  unavailable as well, avoiding a crash from accessing an
  uninitialized mSupportInfo.

test: sailfish boot :D
Change-Id: Ibf43d0bfea9fd6cdeae35248e4dde58c01d3ef3b
Signed-off-by: electimon <electimon@gmail.com>
---
 .../com/android/server/power/hint/HintManagerService.java    | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/services/core/java/com/android/server/power/hint/HintManagerService.java b/services/core/java/com/android/server/power/hint/HintManagerService.java
index 9174855b6..4d81b90b6 100644
--- a/services/core/java/com/android/server/power/hint/HintManagerService.java
+++ b/services/core/java/com/android/server/power/hint/HintManagerService.java
@@ -333,6 +333,8 @@ public final class HintManagerService extends SystemService {
             } catch (RemoteException e) {
                 throw new IllegalStateException("Could not contact PowerHAL!", e);
             }
+        } else {
+            mSupportInfo = getDummySupportInfo();
         }
         if (mSupportInfo.headroom.isCpuSupported) {
             mCpuHeadroomCache = new HeadroomCache<>(2, mSupportInfo.headroom.cpuMinIntervalMillis);
@@ -373,7 +375,10 @@ public final class HintManagerService extends SystemService {
         } catch (RemoteException e) {
             throw new IllegalStateException("Could not contact PowerHAL!", e);
         }
+        return getDummySupportInfo();
+    }

+    private SupportInfo getDummySupportInfo() {
         SupportInfo supportInfo = new SupportInfo();
         supportInfo.usesSessions = isHintSessionSupported();
         // Global boosts & modes aren't currently relevant for HMS clients
--
2.43.0

